packages/react-server-cli/src/compileClient.js

```diff
@@ -0,0 +1,216 @@
+		return `unwrapEs6Module(require("${path.relative(workingDirAbsolute, path.resolve(routesDir, middlewareRelativePath))}"))`
+	}) : [];
+	routesOutput.push(`
+function unwrapEs6Module(module) { return module.__esModule ? module.default : module }
+var coreJsMiddleware = unwrapEs6Module(require('react-server-cli/target/coreJsMiddleware'));
+var coreCssMiddleware = unwrapEs6Module(require('react-server-cli/target/coreCssMiddleware'));
+module.exports = {
+	middleware:[
+		coreJsMiddleware(${JSON.stringify(staticUrl)}),
+		coreCssMiddleware(${JSON.stringify(staticUrl)}),
+		${existingMiddleware.join(",")}
+	],
+	routes:{`);
+
+	for (let routeName of Object.keys(routes.routes)) {
+		let route = routes.routes[routeName];
+
+		routesOutput.push(`
+		${routeName}: {`);
+		for (let name of ["path", "method"]) {
+			routesOutput.push(`
+			${name}: "${route[name]}",`);
+		}
+
+		let formats = normalizeRoutesPage(route.page);
+		routesOutput.push(`
+			page: {`);
+		for (let format of Object.keys(formats)) {
+			const formatModule = formats[format];
+			var relativePathToPage = path.relative(workingDirAbsolute, path.resolve(routesDir, formatModule));
+			routesOutput.push(`
+				${format}: function() {
+					return {
+						done: function(cb) {`);
+			if (isClient) {
+				routesOutput.push(`
+							require.ensure("${relativePathToPage}", function() {
+								cb(unwrapEs6Module(require("${relativePathToPage}")));
+							});`);
+			} else {
+				routesOutput.push(`
+							cb(unwrapEs6Module(require("${relativePathToPage}")));`);
+			}
+			routesOutput.push(`
+						}
+					};
+				},`);
+		}
+		routesOutput.push(`
+			},
+		}`);
+	}
+	routesOutput.push(`
+	}
+};`);
+
+	const routesFilePath = `${workingDirAbsolute}/routes_${isClient ? "client" : "server"}.js`;
+	fs.writeFileSync(routesFilePath, routesOutput.join(""));
+
+	return routesFilePath;
+};
+
+
+// the page value for routes.routes["SomeRoute"] can either be a string for the default
+// module name or an object mapping format names to module names. This method normalizes
+// the value to an object.
+const normalizeRoutesPage = (page) => {
+	if (typeof page === "string") {
+		return {default: page};
+	}
+	return page;
+}
+
+// writes out a bootstrap file for the client which in turn includes the client
+// routes file. note that outputDir must be the same directory as the client routes
+// file, which must be named "routes_client".
+const writeClientBootstrapFile = (outputDir) => {
+	var outputFile = outputDir + "/entry.js";
+	fs.writeFileSync(outputFile, `
+		if (typeof window !== "undefined") {
+			window.__setReactServerBase = function(path) {
+				// according to http://webpack.github.io/docs/configuration.html#output-publicpath
+				// we should never set __webpack_public_path__ when hot module replacement is on.
+				if (!module.hot) {
+					__webpack_public_path__ = path;
+					window.__reactServerBase = path;
```

Comment:
See above note; I think I was wrong in the case where `compileOnly` is used to make a compiled client bundle that can be deployed at any URL. Sigh.